parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/state/ShardManager.java

```diff
@@ -92,6 +96,9 @@ public class ShardManager<K, V> {
     @Getter(AccessLevel.PACKAGE) // visible for testing
     private final NavigableSet<WorkContainer<?, ?>> retryQueue = new TreeSet<>(retryQueueWorkContainerComparator);
 
+    @Getter
+    private final AtomicLong retryItemCnt = new AtomicLong(0);
```

Comment:
Any benefit to use `retryItemCnt` over the `retryQueue.size()` ? 
retryQueue is a TreeSet backed by TreeMap - which has `size` variable - that it keeps updated on changes to the TreeMap - so it should be similarly performant as its a cached value and doesn't require scan of the queue.

In addition removal of  `retryItemCnt` and using `retryQueue.size()` directly means code tracking changes to `retryQueue` to update the `retryItemCnt` is unnecessary - which is a plus.
For example below:
```
 if (this.retryQueue.remove(removedWC)) {
                retryItemCnt.decrementAndGet();
            }
 ```
can be reverted back to just 
```
this.retryQueue.remove(removedWC);
```

Reply:
fixed